### PR TITLE
[enhancement](Nereids) optimize bloom filter size computing strategy

### PR DESCRIPTION
**Pull request by morrySnow**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by morrySnow** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/processor/post/RuntimeFilterGenerator.java [hunk 1]
```diff
@@ -111,8 +114,21 @@ public PhysicalPlan visitPhysicalHashJoin(PhysicalHashJoin<? extends Plan, ? ext
...
+                    if (right.getStats() != null) {
+                        List<Double> ndvs = join.getHashJoinConjuncts().stream()
+                                .map(Expression::getInputSlots)
+                                .flatMap(Set::stream)
+                                .filter(s -> right.getOutputExprIdSet().contains(s.getExprId()))
+                                .map(s -> right.getStats().columnStatistics().get(s))
+                                .filter(Objects::nonNull)
+                                .map(cs -> cs.ndv)
+                                .collect(Collectors.toList());
+                        buildSideNdv = (long) StatsMathUtil.multiNdv(ndvs);
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/processor/post/RuntimeFilterGenerator.java:128]: buildSideNdv should be less than build side row count.
buildSideNdv = Math.max(buildSideNdv, rowCount)

File: fe/fe-core/src/main/java/org/apache/doris/nereids/stats/StatsMathUtil.java [hunk 2]
```diff
@@ -59,4 +64,21 @@ public static double divide(double a, double b) {
         return a / nonZeroDivisor(b);
     }
 
+    /**
+     * compute the multi columns unite ndv
+     */
+    public static double multiNdv(List<Double> ndvs) {
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/stats/StatsMathUtil.java:70]: how about rename to accumulativeNdv or jointNdv ?

**Review by englefly — DISMISSED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/stats/StatsMathUtil.java [hunk 3]
```diff
@@ -59,4 +64,21 @@ public static double divide(double a, double b) {
...
+        if (CollectionUtils.isEmpty(ndvs)) {
+            return -1;
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/stats/StatsMathUtil.java:72]: if ndvs.isEmpty(), return 0 is resonable.

File: fe/fe-core/src/main/java/org/apache/doris/nereids/processor/post/RuntimeFilterGenerator.java (hunk 1 above)
Comment (morrySnow) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/processor/post/RuntimeFilterGenerator.java:128]: done

File: fe/fe-core/src/main/java/org/apache/doris/nereids/stats/StatsMathUtil.java (hunk 2 above)
Comment (morrySnow) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/stats/StatsMathUtil.java:70]: done

**Comment by morrySnow** [comment]
run buildall

**Review by Gabriel39 — APPROVED** [review]
LGTM

**Pull request merged**